examples/seismic/elastic/wavesolver.py

```diff
@@ -0,0 +1,100 @@
+from devito import TimeFunction, memoized_meth, warning
+from examples.seismic import Receiver
+from examples.seismic.elastic.operators import ForwardOperator
+
+
+class ElasticWaveSolver(object):
+    """
+    Solver object that provides operators for seismic inversion problems
+    and encapsulates the time and space discretization for a given problem
+    setup.
+
+    :param model: Physical model with domain parameters
+    :param source: Sparse point symbol providing the injected wave
+    :param receiver: Sparse point symbol describing an array of receivers
+    :param time_order: Order of the time-stepping scheme (default: 2, choices: 2,4)
+                       time_order=4 will not implement a 4th order FD discretization
+                       of the time-derivative as it is unstable. It implements instead
+                       a 4th order accurate wave-equation with only second order
+                       time derivative. Full derivation and explanation of the 4th order
+                       in time can be found at:
+                       http://www.hl107.math.msstate.edu/pdfs/rein/HighANM_final.pdf
+    :param space_order: Order of the spatial stencil discretisation (default: 4)
+
+    Note: space_order must always be greater than time_order
```

Comment:
is this checked?

Reply:
Old leftover, it's everywhere in acoustic and tti too.